utils/src/activityLog/activities/ExecuteActivity.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TreeItemCollapsibleState } from "vscode";
+import * as types from '../../../index';
+import { AppResource } from "../../../unified";
+import { AzExtParentTreeItem } from "../../tree/AzExtParentTreeItem";
+import { GenericTreeItem } from "../../tree/GenericTreeItem";
+import { nonNullProp } from "../../utils/nonNull";
+import { ActivityBase } from "../Activity";
+
+
+interface ExecuteActivityData<C extends types.IActionContext> {
+    title: string;
+    context: C;
+}
+
+export class ExecuteActivity<C extends types.IActionContext> extends ActivityBase<void> {
```

Comment:
nturinski:
> Would it make sense to make an ActivityContext that has activityResult as a defined property?
> 
> It might easier to define a lot of the state and stuff as well on the context so it's easier to pass around instead of having it all live on the Activity itself.

Reply:
Nathan's other comment was basically the same thing as mine. :smile:

Reply:
Thanks for copying all of these over

Reply:
I made `ExecuteActivityContext`

```ts
export declare interface ExecuteActivityContext {
    registerActivity: (activity: Activity) => Promise<void>;
    /**
     * Becomes label of activity tree item, defaults to wizard title or "Azure Activity"
     */
    activityTitle?: string;
    /**
     * Set to show a "Click to view resource" child on success.
     */
    activityResult?: AppResource;
}
```

Reply:
> Nathan's other comment was basically the same thing as mine. 😄

Great minds? 🧠🧠